CHANGELOG.md

```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - Improve type inference around unsigned integer literals.
+- Improve type inference around array constructors containing integer literals.
```

Comment:
Could probably be combined with the previous entry.

Reply:
The array constructors type inference issues were a distinct problem from what was fixed with the integer type inference.
The problem with array constructors was the way we determined which element "won" and became the element type for the resulting dynamic array. We were doing a dumb size comparison, without consideration for signedness.